python/ray/_private/log_rotation_with_pipe.py

```diff
@@ -0,0 +1,119 @@
+"""Implement log rotation with pipe."""
+
+from typing import IO, AnyStr
+import sys
+import os
+import logging
+import threading
+import atexit
+from logging.handlers import RotatingFileHandler
+
+
+class PipeStreamWriteHandle(IO[AnyStr]):
+    def __init__(self, write_fd: int, log_name: str):
+        self.name = log_name
+        self.write_fd = write_fd
+        self.stream = os.fdopen(write_fd, "w")
+
+    def name(self):
+        return self.name
+
+    def fileno(self):
+        return self.write_fd
+
+    def write(self, s: str) -> int:
+        return self.stream.write(s)
+
+    def flush(self):
+        self.stream.flush()
+
+    def close(self):
+        self.stream.close()
+
+    def isatty(self) -> bool:
+        return False
+
+    def readable(self) -> bool:
+        return False
+
+    def writable(self) -> bool:
+        return True
+
+    def seekable(self) -> bool:
+        return False
+
```

Comment:
`configure_log_file` only requires this object has `fileno()` method not others?

Reply:
write, flush, name, fileno is definitely needed --- I have met issues without any of them;
Others are suggested by gpt, didn't verify one by one.

My personal opinion is it doesn't hurt to implement these overloads; 
- All these overload functions are straightforward and short enough
- Some attribute function help explain the class, i.e. readable and writable.